content/05-mysql/11-tools/01-mysql-config-editor.mdx

```diff
@@ -0,0 +1,271 @@
+ Connection:			Localhost via UNIX socket
+ Server characterset:	utf8mb4
+ Db     characterset:	utf8mb4
+ Client characterset:	utf8mb4
+ Conn.  characterset:	utf8mb4
+ UNIX socket:			/var/run/mysqld/mysqld.sock
+ Binary data as:		Hexadecimal
+ Uptime:				1 day 21 hours 37 min 49 sec
+ 
+ Threads: 2  Questions: 66  Slow queries: 0  Opens: 186  Flush tables: 3  Open tables: 105  Queries per second avg: 0.000
+ --------------
+```
+
+You can see the current user under `Current user`, but you can also view the details about how you are connected to the server by looking at the `Connection` and `UNIX socket` items.
+
+## Displaying available login paths
+
+While the connection details you define are stored in a file called `.mylogin.cnf` in your home directory, the contents are encrypted for security.  To view the configured information, you need to use the `mysql_config_editor` again.
+
+To view the default login information you've configured, which is stored under the `client` login path, you can use the `print` subcommand without any additional options:
+
+```
+mysql_config_editor print
+```
+```
+[client]
+user = "root"
+password = *****
+```
+
+MySQL uses an [INI style file format](https://en.wikipedia.org/wiki/INI_file) to group connection details under the appropriate login path label.  You may also notice that the password is obscured.  This, again, is a security measure so as not to leak the saved password.
+
+To view a different login path, you can supply the `--login-path=` option as usual:
+
+```
+mysql_config_editor print --login-path=testing
+```
+```
+[testing]
+user = "testuser"
+password = *****
+host = "dev.example.com"
+port = 5555
+```
+
+To show all of the configured login paths, you can add the `--all` flag instead:
+
+```
+mysql_config_editor print --all
+```
+```
+[sales]
+user = "salesadmin"
+password = *****
+[testing]
+user = "testuser"
+password = *****
+host = "dev.example.com"
+port = 5555
+[client]
+user = "root"
+password = *****
+```
+
+## Removing connection information
+
+You can remove the connection information associated with a login path with the `remove` subcommand.  Providing the `--login-path` will allow `mysql_config_editor` to target the appropriate entry.
+
+For example, to remove the connection information for the `sales` login path, you can type:
+
+```
+mysql_config_editor remove --login-path=sales
+```
+
+If you check the configured entries, you will find that the `sales` login path has been removed:
+
+```
+mysql_config_editor print --all
+```
+```
+[testing]
+user = "testuser"
+password = *****
+host = "dev.example.com"
+port = 5555
+[client]
+user = "root"
+password = *****
+```
+
+You can also remove a specific parameter from the login path's connection information.  For instance, if the MySQL server at "dev.example.com" has been reconfigured to now run on the default 3306 port, you can remove the port information.  To do so, you'd provide the `--port` flag to along after the `--login-path=`:
```

Comment:
I am not sure if I am reading the last sentence incorrectly or don't know the phrasing, but "flag to along after the" seems incorrect. I apologize if I am mistaken :)

Reply:
Yeah, that's 100% nonsense lol.  Thanks for catching that.